src/fe-text/gui-entry.c

```diff
@@ -1042,3 +1100,31 @@ void gui_entry_redraw(GUI_ENTRY_REC *entry)
 	gui_entry_fix_cursor(entry);
 	gui_entry_draw(entry);
 }
+
+void gui_entry_set_color(GUI_ENTRY_REC *entry, int pos, int len, int color)
```

Comment:
I think we should add a couple of checks to make sure `pos` and `len` are `>= 0` to avoid some potential memory corruption bugs due to careless scripts.